### PR TITLE
Add an option to close pending email reminder dialog

### PR DESCRIPTION
**Pull request by FixRM**
It can be shown for admin users and block the test script from passing

### CODE REVIEW

**Comment by FixRM** [comment]
Hello? Is this repository alive?

**Comment by dtu11** [comment]
Yes the repo is alive. :)  We've been focused on other items and slowly working through issues and feature requests.   

I just want to say thanks for the contribution.   I'm reviewing your changes now and will submit feedback.

**Comment by FixRM** [comment]
Thanks @dtu11

File: Microsoft.Dynamics365.UIAutomation.Api/Browser.cs [hunk 1]
```diff
@@ -175,6 +175,12 @@ public void GoToXrmUri(Uri xrmUri)
         public Office365Navigation Office365Navigation => this.GetPage<Office365Navigation>();
 
         #endregion Office365
+
+        #region Pending Email Reminder
+
```
Comment (dtu11) [inline Microsoft.Dynamics365.UIAutomation.Api/Browser.cs]: I believe we have a Dialogs page that has a CloseWarningDialog method that will handle the Pending Email warning.  I would like to be consistent with how we handle Dialogs and not introduce complexity of individual pages for each.   Can we test the existing CloseWarningDialog or perhaps move the methods to the Dialogs page?

File: Microsoft.Dynamics365.UIAutomation.Api/Pages/PendingEmailReminder.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        /// </summary>
+        /// <example>xrmBrowser.PendingEmail.Close();</example>
+        public BrowserCommandResult<bool> Close(int thinkTime = Constants.DefaultThinkTime)
+        {
+            this.Browser.ThinkTime(thinkTime);
+            return this.Execute(GetOptions("Close Pending Email Reminder"), driver =>
+            {
+                bool returnValue = false;
+
+                if (IsShown)
```
Comment (dtu11) [inline Microsoft.Dynamics365.UIAutomation.Api/Pages/PendingEmailReminder.cs]: For the IsShown we should use a driver.WaitUntilAvailable with a Timespan of 1-2 seconds.  If there is some delay on the client where the dialog is not shown right away then it will impact everything after this execution.

File: Microsoft.Dynamics365.UIAutomation.Api/Browser.cs (hunk 1 above)
Comment (FixRM) [reply Microsoft.Dynamics365.UIAutomation.Api/Browser.cs]: I thought about it but decided to put this methods to separate page for some reason I can't remember. I'll try CloseWarningDialog and come back with comments.

File: Microsoft.Dynamics365.UIAutomation.Api/Pages/PendingEmailReminder.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+                    driver.WaitUntilClickable(By.Id(Elements.ElementId[Reference.PendingEmail.ButBegin]), new TimeSpan(0, 0, 5), d =>
+                    {
+                        var element = d.FindElement(By.Id(Elements.ElementId[Reference.PendingEmail.ButBegin]));
+                        element.Click(true);
+
+                        returnValue = true;
+                    });
+
+                    driver.SwitchTo().DefaultContent();
```
Comment (dtu11) [inline Microsoft.Dynamics365.UIAutomation.Api/Pages/PendingEmailReminder.cs]: This is not really needed.   When an API is executed the switch is done in the constructor so the page itself should know which content needs to be loaded.   This line won't affect anything functionally but it just adds an additional switch and affects to the API's execution time which is tracked.

File: Microsoft.Dynamics365.UIAutomation.Api/Pages/PendingEmailReminder.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            return this.Execute(GetOptions("Close Pending Email Reminder"), driver =>
+            {
+                bool returnValue = false;
+
+                if (IsShown)
+                {
+                    var inlineDialog = driver.FindElement(By.Id(Elements.ElementId[Reference.PendingEmail.InlineDialog]));
+                    driver.SwitchTo().Frame(inlineDialog);
+
+                    driver.WaitUntilClickable(By.Id(Elements.ElementId[Reference.PendingEmail.ButBegin]), new TimeSpan(0, 0, 5), d =>
```
Comment (dtu11) [inline Microsoft.Dynamics365.UIAutomation.Api/Pages/PendingEmailReminder.cs]: You can switch this to the ClickWhenAvailable which uses the WaitUntilClickable.  I just noticed that the Click action, in ClickWhenAvailable, doesn't use the ignoreStaleElementExceptions option we could update that.

File: Microsoft.Dynamics365.UIAutomation.Browser/BrowserDriverFactory.cs [hunk 5]
```diff
@@ -52,7 +52,7 @@ private static IWebDriver CreateLocalDriver(BrowserOptions options)
                 case BrowserType.Firefox:
```
Comment (dtu11) [inline Microsoft.Dynamics365.UIAutomation.Browser/BrowserDriverFactory.cs:52]: I like this additional functionality.  Any chance we can separate these changes into it's own PR?   I would like to get this in while we work through the ClosePendingEmail dialog changes.

Comment (FixRM) [reply Microsoft.Dynamics365.UIAutomation.Browser/BrowserDriverFactory.cs:52]: Yes I did it accidentally (I'm a little bit new here :) ).

Comment (FixRM) [reply Microsoft.Dynamics365.UIAutomation.Browser/BrowserDriverFactory.cs:52]: @dtu11 Done: #240

File: Microsoft.Dynamics365.UIAutomation.Api/Browser.cs (hunk 1 above)
Comment (FixRM) [reply Microsoft.Dynamics365.UIAutomation.Api/Browser.cs]: Yes you are right. CloseWarningDialog works as expected so this PR is not needed

**Comment by FixRM** [comment]
Thank you for your time. CloseWarningDialog works well, this PR is not needed any more
